.github/workflows/pr-build-image.yaml

```diff
@@ -43,6 +43,17 @@ jobs:
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
 
+      - name: Check Image and Relevant Changes
+        id: check-image
+        uses: ./.github/actions/check-image-and-changes
+
+      - name: Check if build should be skipped
```

Comment:
a further simplication could be to move this check into the action itself.

then the output of the `check-image-and-changes` is not just two booleans, but an exit code of 0 if the job can be terminated right there. 

Of course your logged message would need to be something like

```
            echo "::notice::Skipping job: Image already exists and no relevant changes detected."
```